### PR TITLE
Listing, MapLocation, User objects

### PR DESCRIPTION
**Pull request by LindseyWeiskopf**
Summary: Created the Listing, MapLocation, and user object classes

Listing class: This class contains the various attributes that will be needed for each listing on the webapp: name, location, image, ratings, and website. Right now, the class just contains accessor methods for all of the instance variables.

MapLocation class: This class essentially contains a coordinate (lat and long) to be able to track users and listings.

User class: As of right now (Signed out user), this class contains an overloaded constructor so that a user can be initialized with a MapLocation (result of Geolocation API/coordinate) or a zip code.

Thanks!

### CODE REVIEW

File: src/main/java/com/google/sps/data/Listing.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  * @return Listing object
+  **/
+
+  public Listing(String name, MapLocation mapLocation, int rating, String image, String url) {
+    this.name = name;
+    this.mapLocation = mapLocation;
+    this.rating = rating;
+    this.image = image;
+    this.url = url;
+  }
```
Comment (ethanimooney) [inline src/main/java/com/google/sps/data/Listing.java:27]: I might add a section for a business description. Not sure if we can pull that from the Places API in the long run, do you?

File: src/main/java/com/google/sps/data/Listing.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
+package com.google.sps.data;
+
+/** Class for Listing object with the specified attributes returned from the places API **/
+public class Listing{
```
Comment (ethanimooney) [inline src/main/java/com/google/sps/data/Listing.java:4]: Super, mega, big, important nit .... space before the bracket :)

File: src/main/java/com/google/sps/data/Listing.java (hunk 1 above)
Comment (LindseyWeiskopf) [reply src/main/java/com/google/sps/data/Listing.java:30]: The Places API does not return a business description, but it does return a "type" like food and establishment which signifies that the entity is a business which might be even more helpful down the road. I added a businessType parameter to account for that. I didn't see anything regarding business description on the places API, but that's definitely something we can add in the future if we find a way to get that info.

File: src/main/java/com/google/sps/data/Listing.java (hunk 2 above)
Comment (LindseyWeiskopf) [reply src/main/java/com/google/sps/data/Listing.java]: Ugh! How could I miss that!!

File: src/main/java/com/google/sps/data/Listing.java (hunk 1 above)
Comment (ethanimooney) [reply src/main/java/com/google/sps/data/Listing.java:30]: Works for me, thx for the explanation

File: src/main/java/com/google/sps/data/Listing.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
+package com.google.sps.data;
+
+/** Class for Listing object with the specified attributes returned from the places API **/
```
Comment (shaquille-hall) [inline src/main/java/com/google/sps/data/Listing.java:3]: nit: You can remove "Class for".

File: src/main/java/com/google/sps/data/Listing.java (hunk 1 above)
Comment (shaquille-hall) [reply src/main/java/com/google/sps/data/Listing.java:30]: +1

File: src/main/java/com/google/sps/data/MapLocation.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.google.sps.data;
+
+/** Class representing a MapLocation as a coordinate **/
```
Comment (shaquille-hall) [inline src/main/java/com/google/sps/data/MapLocation.java:3]: nit: You can reduce this to "MapLocation representing coordinate values"

File: src/main/java/com/google/sps/data/User.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package com.google.sps.data;
+
+/** Class representing a user with a location for v1 **/
```
Comment (shaquille-hall) [inline src/main/java/com/google/sps/data/User.java:3]: nit: You can remove the note that it's a class.

File: src/main/java/com/google/sps/data/Listing.java (hunk 3 above)
Comment (LindseyWeiskopf) [reply src/main/java/com/google/sps/data/Listing.java]: Changed!

File: src/main/java/com/google/sps/data/MapLocation.java (hunk 4 above)
Comment (LindseyWeiskopf) [reply src/main/java/com/google/sps/data/MapLocation.java]: Got it

**Pull request merged**